egg/core/language_analysis.py

```diff
@@ -263,6 +289,72 @@ def on_test_end(self, loss, logs, epoch):
             self.print_message(logs, "test", epoch)
 
 
+class BosDisent(Callback):
+    """
+    Bag-of-symbols disentanglement metric, introduced in "Compositionality and Generalization in Emergent Languages",
+    Chaabouni et al., ACL 2020.
+    """
+
+    def __init__(
+        self,
+        vocab_size: int,
+        print_train: bool = False,
+        print_test: bool = True,
+        is_gumbel: bool = False,
+    ):
+        super().__init__()
+        assert (
+            print_train or print_test
+        ), 'At least on of "print_train" and "print_train" must be enabled'
+
+        self.vocab_size = vocab_size
+        self.print_train = print_train
+        self.print_test = print_test
+        self.is_gumbel = is_gumbel
+
+    @staticmethod
+    def bosdis(attributes: torch.Tensor, messages, vocab_size) -> float:
```

Comment:
optionally this could be joined with PosDis class into some Dis thingy, wdyt? it should be the same modulo the histogram calculation

Reply:
If it computing MI values is efficient then having both metrics returned by default would be a good idea. Otherwise we could set both optional (with one to be specified)